state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -644,90 +644,88 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		// Transactional ensures that any failed swap will rollback all storage changes.
+		#[transactional]
 		fn process_swaps_for_block(block: BlockNumberFor<T>) -> Result<(), BatchExecutionError> {
-			// Wrap the entire swapping section as a transaction, any failed swap will rollback all
-			// storage changes.
-			with_storage_layer(|| -> Result<(), BatchExecutionError> {
-				let mut swaps = SwapQueue::<T>::take(block);
-
-				// Swap into Stable asset first.
-				Self::do_group_and_swap(&mut swaps, SwapLeg::ToStable)?;
-
-				// Take NetworkFee for all swaps
-				for swap in swaps.iter_mut() {
-					debug_assert!(
-						swap.stable_amount.is_some(),
-						"All swaps should have Stable amount set here"
-					);
-					let stable_amount = swap.stable_amount.get_or_insert_with(Default::default);
-					*stable_amount = T::SwappingApi::take_network_fee(*stable_amount);
-				}
+			let mut swaps = SwapQueue::<T>::take(block);
```

Comment:
I think it makes sense to return Ok(()) early here if there are no swaps in the batch.

Reply:
Updated